api/grpc_server.go

```diff
@@ -310,14 +321,25 @@ func (s SpacemeshGrpcService) SetAwardsAddress(ctx context.Context, id *pb.Accou
 func (s SpacemeshGrpcService) GetMiningStats(ctx context.Context, empty *empty.Empty) (*pb.MiningStats, error) {
 	//todo: we should review if this RPC is necessary
 	log.Info("GRPC GetInitProgress msg")
-	stat, coinbase, dataDir := s.Mining.MiningStats()
-	return &pb.MiningStats{Status: int32(stat), Coinbase: coinbase, DataDir: dataDir}, nil
+	stat, remainingBytes, coinbase, dataDir := s.Mining.MiningStats()
+	return &pb.MiningStats{
+		DataDir:        dataDir,
+		Status:         int32(stat),
+		Coinbase:       coinbase,
+		RemainingBytes: remainingBytes,
+	}, nil
 }
 
-func (s SpacemeshGrpcService) GetTotalAwards(ctx context.Context, empty *empty.Empty) (*pb.SimpleMessage, error) {
-	//todo: we should review if this RPC is necessary
-	log.Info("GRPC GetTotalAwards msg")
-	return &pb.SimpleMessage{Value: "1234"}, nil
+func (s SpacemeshGrpcService) GetNodeStatus(context.Context, *empty.Empty) (*pb.NodeStatus, error) {
+	return &pb.NodeStatus{
+		Peers:         s.PeerCounter.PeerCount(),
+		MinPeers:      uint64(s.Config.P2P.OutboundPeersTarget),
```

Comment:
currently this config entry isn't used, what's used is `config.P2P.SwarmConfig.RandomConnections` I guess eventually I'll reorganize p2p config and we'll use this one. so change both the above and below to RandomConnections